draft-stark-expect-ct.md

```diff
@@ -23,6 +23,316 @@ TODO: Wite an abstract here
+Expect-CT response header field over an error-free TLS connection, inluding the
+validation added in {{expect-ct-compliance}}, that included the `enforce`
+directive.
+
+To note a host as an Expect-CT host, the UA MUST set its Expect-CT metadata to
+the effectie expiration date and report-uri (if any) given in the most recently
+received valid Expect-CT header.
+
+For forward compatibility, the UA MUST ignore any unrecognized Expect-CT header
+directives, while still processing those directives it does
+recognize. {{response-header-field-syntax} specifies the directives `enforce`,
+`max-age`, and `report-uri`, but future specifications and implementations might
+use additional directives.
+
+## Evaluating Expect-CT Connections for CT Compliance {#expect-ct-compliance}
+
+When a UA connects to an Expect-CT host using a TLS connection, if the TLS
+connection has errors, the UA MUST terminate the connection without allowing the
+user to proceed anyway. (This behavior is the same as that required by
+{{!RFC6797}}.)
+
+If the connection has no errors, then the UA will apply an additional
+correctness check: compliance with a CT policy. A UA should evaluate compliance
+with its CT policy whenever connecting to an Expect-CT host, as soon as
+possible. It is acceptable to skip this CT compliance check for some hosts
+according to local policy. For example, a UA may disable CT compliance checks
+for hosts whose validated certificate chain terminates at a user-defined trust
+anchor, rather than a trust anchor built-in to the UA (or underlying platform).
+
+SOMETHING SOMETHING about the UA getting to choose the policy based on the
+SCTs. Example policies in an appendix? Something advising UAs to publish their
+policies somewhere? Something about using reporting to discover policy changes?
+
+A UA that has previously noted a host as an Expect-CT host MUST evaluate
+evaluate CT compliance when setting up the TLS session, before beginning an HTTP
+conversation over the TLS channel.
+
+If the UA does not evaluate CT compliance, e.g. because the user has elected to
+disable it, or because a presented certificate chain chains up to a user-defined
+trust anchor, UAs SHOULD NOT send Expect-CT reports.
+
+# Interactions with Preloaded Pin Lists
+
+# Reporting Expect-CT Failure
+
+When the UA receives an Expect-CT header with a `report-uri` directive that does
+not comply with the UA's CT policy, or when the UA connects to a noted Expect-CT
+host that does not comply with the CT policy, the UA SHOULD report Expect-CT
+failures to the configured `report-uri`. The UA does this by POSTing a JSON
+{{!RFC7159}} message to the URI. The JSON message takes this form:
+
+~~~
+{
+FILL IN REPORT FORMAT HERE
+}
+~~~
+
+Whitespace outside of quoted strings is not significant. The key/value pairs may
+appear in any order, but each MUST appear only once.
+
+EXPLAIN report fields here
+
 # Security Considerations
 
+## Maximum max-age
+
+1 year?
+
+# Privacy Considerations
+
```

Comment:
If JS attempts an async load of a HTTPS resource that has expect-ct set, you can determine if the evaluating user agent has been to that site recently.

If there's a `report-uri`, at least the report-receiver could notice this behavior, though.